core/frontend/src/main/scala/io/udash/view/ViewRenderer.scala

```diff
@@ -15,19 +13,27 @@ private[udash] class ViewRenderer(rootElement: => Element) {
   private lazy val endpoint = rootElement
   private val views = mutable.ArrayBuffer[View]()
 
-  private def mergeViews(path: List[View]): View = {
+  private def renderChild(parent: View, child: Option[View]): Unit =
+    parent match {
+      case p: ContainerView =>
+        p.renderChild(child)
+      case _ =>
```

Comment:
It should be possible to use a more complicated data structure than a list/buffer to avoid this case and traverse a statically checked view tree.